### PR TITLE
Add Get, List Extension Types

### PR DESCRIPTION
**Pull request by carlocardella**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [x] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [ ] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+      "description": "The Azure subscription ID. This is a GUID-formatted string (e.g. 00000000-0000-0000-0000-000000000000)"
+    },
+    "ResourceGroupNameParameter": {
+      "name": "resourceGroupName",
+      "in": "path",
+      "required": true,
+      "type": "string",
+      "description": "The name of the resource group.",
+      "x-ms-parameter-location": "method"
+    },
```
Comment (safeermohammed) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:281]: remomve this params as these are taken from common

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/GetExtensionTypes.json [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      "body": {
+        "id": "/subscriptions/subId1/resourceGroups/rg1/providers/Microsoft.Kubernetes/connectedClusters/clusterName1/providers/Microsoft.KubernetesConfiguration/extensionTypes",
+        "version": "1.0.0",
+        "releaseTrains": "preview",
+        "supportedScopes": {
+          "defaultScope": "cluster",
+          "clusterScopeSettings": {
+            "AllowMultipleInstances": false,
+            "defaultReleaseNamespace": "kube-system"
+          }
```
Comment (safeermohammed) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/GetExtensionTypes.json:23]: For all examples, these go inside properties bag

File: package-lock.json [hunk 3]
```diff
@@ -1,8 +1,8147 @@
 {
   "name": "azure-rest-api-specs-tests",
   "version": "0.1.0",
-  "lockfileVersion": 1,
```
Comment (jianyexi) [inline package-lock.json:4]: could you revert this change, as it's unrelated

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/kubernetesconfiguration.json [hunk 4]
```diff
@@ -1,724 +0,0 @@
-{
-  "swagger": "2.0",
-  "info": {
```
Comment (jianyexi) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/kubernetesconfiguration.json:3]: why remove this file ?

Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/kubernetesconfiguration.json:3]: There are no changes for `kubernetesconfiguration`, should I include it anyway?

File: package-lock.json (hunk 3 above)
Comment (carlocardella) [reply package-lock.json:4]: Will do in the next update

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 5]
```diff
@@ -0,0 +1,424 @@
...
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/{clusterRp}/{clusterType}/locations/{location}/extensionTypes/{extensionTypeName}/versions": {
+      "get": {
+        "description": "List available versions for an Extension Type",
+        "operationId": "ExtensionTypeVersions_List",
+        "x-ms-examples": {
+          "ExtensionTypeVersions_List": {
```
Comment (zcxcasd) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:138]: I'm not understanding this one... if clusterRp is not the kubernetesconfiguration rp, how does kubernetesconfiguration own the extension type/versions?

Comment (zcxcasd) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:138]: did you mean to write /subscriptions/{subscriptionId}/providers/{clusterRp}/{clusterType}/locations/{location}/providers/Microsoft.KubernetesConfiguration/extensionTypes/{extensionTypeName}" or something similar?

---
In reply to: [608965251](https://github.com/Azure/azure-rest-api-specs/pull/13562#discussion_r608965251) [](ancestors = 608965251)

Comment (varunkumta) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:138]: the url is missing multiple segments like the clusterName and providers/Microsoft.KubernetesConfiguration

---
In reply to: [608965251](https://github.com/Azure/azure-rest-api-specs/pull/13562#discussion_r608965251) [](ancestors = 608965251)

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/LocationExtensionTypes_List.json [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+    "subscriptionId": "00000000-0000-0000-0000-000000000000",
+    "clusterRp": "Microsoft.Kubernetes",
+    "clusterType": "connectedClusters",
+    "location": "eastus",
+    "api-version": "2021-05-01-preview"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "id": "/subscriptions/00000000-0000-0000-0000-000000000000/providers/connectedClusters/locations/eastus/extensionTypes",
```
Comment (varunkumta) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/LocationExtensionTypes_List.json:12]: not a valid id .. please fix in all files

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 7]
```diff
@@ -0,0 +1,424 @@
...
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{clusterRp}/{clusterType}/{clusterName}/providers/Microsoft.KubernetesConfiguration/extensionTypes/{extensionTypeName}": {
```
Comment (varunkumta) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:37]: >/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/{clusterRp}/{clusterType}/{clusterName} [](start = 5, length = 116)

add x-ms-pageable even if you don't foresee the need for paging right now

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 8]
```diff
@@ -0,0 +1,424 @@
...
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/{clusterRp}/{clusterType}/locations/{location}/extensionTypes/{extensionTypeName}/versions": {
```
Comment (varunkumta) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:133]: you also need to expose /versions/{version}

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 9]
```diff
@@ -0,0 +1,424 @@
...
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/{clusterRp}/{clusterType}/locations/{location}/extensionTypes": {
```
Comment (varunkumta) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:178]: >extensionTypes [](start = 94, length = 14)

also extensionTypes/{extensionTypeName}

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 10]
```diff
@@ -0,0 +1,424 @@
...
+          },
+          {
+            "$ref": "../../../../../common-types/resource-management/v2/types.json#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/ExtensionType"
```
Comment (varunkumta) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:118]: >ExtensionType [](start = 37, length = 13)

this needs to be ExtensionTypeList

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/LocationExtensionTypes_List.json (hunk 6 above)
Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/LocationExtensionTypes_List.json:12]: does it have to be a valid subscription or just a valid guid? 
sorry, I thought this was intended to be a sort of placeholder

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json (hunk 9 above)
Comment (varunkumta) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:178]: also fix url path

---
In reply to: [609067918](https://github.com/Azure/azure-rest-api-specs/pull/13562#discussion_r609067918) [](ancestors = 609067918)

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 11]
```diff
@@ -0,0 +1,424 @@
...
+              "$ref": "#/definitions/ExtensionVersionList"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
```
Comment (varunkumta) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:175]: add x-ms-pageable

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/kubernetesconfiguration.json (hunk 4 above)
Comment (safeermohammed) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/kubernetesconfiguration.json:3]: Do we know why we have this in first place as this is new API version and folder?

---
In reply to: [603394448](https://github.com/Azure/azure-rest-api-specs/pull/13562#discussion_r603394448) [](ancestors = 603394448)

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:37]: unless it's required, I prefer to keep this consistent with the other API versions we have

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/LocationExtensionTypes_List.json (hunk 6 above)
Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/LocationExtensionTypes_List.json:12]: the example files are generated by `oav generate-examples`, I'll review and update `extensionTypes.json`

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json (hunk 8 above)
Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:133]: The array of versions will be similar to this:

Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:133]: the array of versions will be like this: 

```json
"properties": {
   "releaseTrains": [
      "stable": [
         "1.0.0",
         "1.0.1"
      ],
      "preview": [
         "0.1.2-preview",
         "0.1.3-preview "
      ],
   ]
}
```

`/versions/{version}` would return the same info already requested through the route, is it really needed?

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/kubernetesconfiguration.json (hunk 4 above)
Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/kubernetesconfiguration.json:3]: the branch for this API version was created based on an older version, therefore those files were present in this branch as well

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 12]
```diff
@@ -0,0 +1,422 @@
...
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/{clusterRp}/{clusterType}/locations/{location}/extensionTypes/{extensionTypeName}/versions": {
```
Comment (carlocardella) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json]: typo (shoul be `KubernetesConfiguration`), I'll fix it in the next iteration

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 13]
```diff
@@ -0,0 +1,422 @@
...
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.KubernetesConfigurations/locations/{location}/extensionTypes/{extensionTypeName}/versions": {
```
Comment (carlocardella) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:127]: typo (shoul be `KubernetesConfiguration`), I'll fix it in the next iteration

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 14]
```diff
@@ -0,0 +1,422 @@
...
+      "type": "oauth2",
+      "authorizationUrl": "https://login.microsoftonline.com/common/oauth2/authorize",
+      "flow": "implicit",
+      "description": "Azure Active Directory OAuth2 Flow",
+      "scopes": {
+        "user_impersonation": "impersonate your user account"
+      }
+    }
+  },
+  "paths": {
```
Comment (carlocardella) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:36]: Linting validation is failing with this error: 

```
ERROR (OperationsAPIImplementation/R3023/ARMViolation): Operations API must be implemented for '/providers/Microsoft.KubernetesConfiguration/operations'.
    - file:///D:/GitHub/Azure/azure-rest-api-specs/specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:36:2 ($.paths)
```

@NarayanThiru already has `Microsoft.KubernetesConfiguration/operations` in his changes for this same API preview version [here](https://github.com/NarayanThiru/azure-rest-api-specs/blob/a707a972c537d5c1e25192f2425998ce92495538/specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensions.json#L385). We are going to add the new operations I am defining in Narayan's PR

**Comment by carlocardella** [comment]
@varunkumta what happens if this PR gets merged but our service is not ready yet? In other words, is it possible to create SDKs only for certain resource types?

**Comment by carlocardella** [comment]
@ArcturusZhang could you please help me resolve the errors for the Powershell and Python SDKs? The error is not clear to me, I've been trying to find a good reference in the docs without luck

**Comment by ArcturusZhang** [comment]
Unfortunately python pipeline failure is a known issue and fixing in progress. @dolauli could you help on the powershell failure here?

**Comment by carlocardella** [comment]
@ArcturusZhang, @dolauli, if you guys do not have other comments about the PR content, would it be possible to skip those failing tests since it's a known pipeline issue not depending on my changes?

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json [hunk 15]
```diff
@@ -0,0 +1,411 @@
...
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "../../../../../common-types/resource-management/v2/types.json#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
```
Comment (dolauli) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:167]: According to https://github.com/Azure/autorest/blob/master/docs/extensions/readme.md#x-ms-pageable, it should be changed to 
```
        "x-ms-pageable": {
          "nextLinkName": "nextLink",
          "itemName": "versions"
        }
```

Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/extensionTypes.json:167]: Thanks @dolauli, looks like that did the trick

**Comment by chiragg4u** [comment]
Please go through the contribution checklist for ARM API Review added as part of the review and mark them completed. It will help you fix most of the issues and improve the quality of swagger and help close the PR faster. This is a requirement before review can be continued.

**Comment by carlocardella** [comment]
> Please go through the contribution checklist for ARM API Review added as part of the review and mark them completed. It will help you fix most of the issues and improve the quality of swagger and help close the PR faster. This is a requirement before review can be continued.

Thanks, updated

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/ClusterExtensionTypes_List.json [hunk 16]
```diff
@@ -0,0 +1,46 @@
...
+                  },
+                  "id": "aaaaaaaaaaaaaaaa",
+                  "name": "aaaaaaaaaa",
+                  "type": "aaaaaaaaaaaaaaaaaaa"
+                }
+              }
+            },
+            "systemData": {
+              "createdBy": "aaaaaaaa",
+              "createdByType": "User",
```
Comment (chiragg4u) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/ClusterExtensionTypes_List.json:34]: I assume cluster extensions are always created by SYSTEM and never by USER.

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/ClusterExtensionTypes_List.json [hunk 17]
```diff
@@ -0,0 +1,46 @@
...
+              "createdAt": "2021-04-14T22:44:27.144Z",
+              "lastModifiedBy": "aaaaaaaaa",
```
Comment (chiragg4u) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/ClusterExtensionTypes_List.json:36]: >lastModifiedBy [](http://example.com/codeflow?start=15&length=14)

Since there is no PUT will there be ever modifiedBy?

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/LocationExtensionTypes_List.json [hunk 18]
```diff
@@ -0,0 +1,44 @@
+{
+  "parameters": {
+    "subscriptionId": "aaaaaaaaaa",
+    "location": "aaaaaa",
+    "api-version": "2021-05-01-preview"
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "value": [
```
Comment (chiragg4u) [inline specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/LocationExtensionTypes_List.json:10]: I assume extensions available in a given location is a superset of extensions across all the clusters in that location. And users will be able to select one of the extension on their cluster.

**Comment by chiragg4u** [comment]
I see @Varun has already reviewed the PR extensively. 😊

I’ve some non blocking questions which I’ve put on the PR, approving the PR from ARM side please follow the process given @ https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/212/Swagger-PR-Review to complete and get the PR merged.

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/ClusterExtensionTypes_List.json (hunk 17 above)
Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/ClusterExtensionTypes_List.json:36]: We will have PUT in a future release

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/LocationExtensionTypes_List.json (hunk 18 above)
Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/LocationExtensionTypes_List.json:10]: Correct

File: specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/ClusterExtensionTypes_List.json (hunk 16 above)
Comment (carlocardella) [reply specification/kubernetesconfiguration/resource-manager/Microsoft.KubernetesConfiguration/preview/2021-05-01-preview/examples/ClusterExtensionTypes_List.json:34]: We discussed today, this will likely be the "publisher" e.g. Microsoft vs. eventually third party publishers ("User" here is how the tool generated the sample)

**Review by jianyexi — APPROVED** [review]
